Packs/VersaDirector/Integrations/VersaDirector/VersaDirector_test.py

```diff
@@ -0,0 +1,1751 @@
+    """
+    Given:
+        - Arguments passed: organization, appliance_name
+
+    When:
+        - vd-appliance-user-modified-application-list command is executed
+
+    Then:
+        - The http request is called with the right arguments
+    """
+    from VersaDirector import appliance_user_modified_application_list_command
+
+    class MockResponse:
+        def __init__(self) -> None:
+            self.content = b"""<collection><org>Good</org></collection>"""
+
+    http_request = mocker.patch.object(client, "_http_request", return_value=MockResponse())
+    args = {
+        "organization": "org_name",
+        "appliance_name": "appliance_name",
+    }
+
+    appliance_user_modified_application_list_command(client, args)
+    http_request.assert_called_with(
+        "GET",
+        url_suffix="api/config/devices/device/appliance_name/config/orgs/org-services/org_name/application-identification/application-specific-options/app-specific-option-list",
+        params={},
+        headers={"Content-Type": "application/xml"},
+        resp_type="xml",
+        ok_codes=(200, 201),
+    )
+
+
+def test_predefined_application_list_command(mocker, client):
+    """
+    Given:
+        - Arguments passed: organization
+
+    When:
+        - vd-predefined-application-list command is executed
+
+    Then:
+        - The http request is called with the right arguments
+    """
+    from VersaDirector import predefined_application_list_command
+
+    http_request = mocker.patch.object(client, "_http_request", return_value=None)
+    args = {"organization": "org_name"}
+    predefined_application_list_command(client, args)
+    http_request.assert_called_with(
+        "GET",
+        url_suffix="vnms/spack/predefined?xPath=/predefined/config/predefined-applications/application-identification/applications/application",
+        params={},
+        headers={},
+        resp_type="json",
+        ok_codes=(200, 201, 204),
+    )
+
+
+# HEADING: """ HELPER FUNCTIONS TESTS """
+
+
+@pytest.mark.parametrize("input, expected_output", input_data.check_limit_args)
+def test_check_limit(input, expected_output):
+    """
+    Given:
+        - 'limit' value is passed as argument
+    When:
+        - A command that has the option to choose 'limit' argument is executed
+    Then:
+        - Return None if value is valid
+    """
+    from VersaDirector import check_limit
+
+    assert check_limit(input) == expected_output
+
+
+@pytest.mark.parametrize("input, expected_output", input_data.check_limit_error_args)
+def test_check_limit_fail(input, expected_output):
+    """
+    Given:
+        - Invalid 'limit' argument (not a positive number)
+    When:
+        - A command that has the option to choose 'limit' argument is executed
+    Then:
+        - Raises exception
+    """
+    from VersaDirector import check_limit
+
+    with pytest.raises(DemistoException) as e:
+        check_limit(input)
+    assert str(e.value.message) == expected_output
```

Comment:
The exception is always the same one, you can just write it here and not pass as parameter. Same in other similar tests (on some you already did it).